### PR TITLE
fix interface type decoded as empty value by recursive approach

### PR DESCRIPTION
**Pull request by marco709394**
Found that field with interface type is decoded without value if it does not support "UnmarshalBinary".
I am not very sure about how "putvalue" works but recursively calling "Decode" solved the problem

### CODE REVIEW

**Comment by ineiti** [comment]
Thanks for the bug-report - do you have a test-case we could add to better understand what's going on?

**Comment by ineiti** [comment]
Looks fine to me - I don't know whether that makes our protobuf even more incompatible with the standard or not.

**Comment by Liamsi** [comment]
> Looks fine to me - I don't know whether that makes our protobuf even more incompatible with the standard or not.

Looks fine to me, too. Thanks a lot for your effort @marco709394 
I'll test if this has any side-effect or introduces any incompatibility.

**Comment by nikkolasg** [comment]
+1

**Comment by Liamsi** [comment]
I don't see a problem here. OK to merge.

File: decode.go [hunk 1]
```diff
@@ -308,7 +308,7 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 		// Decode into the object the interface points to.
 		// XXX perhaps better ONLY to support self-decoding
 		// for interface fields?
-		return de.putvalue(wiretype, val.Elem(), v, vb)
+		return Decode(vb, val.Interface())
```
Comment (Liamsi) [inline decode.go]: Wait, if we ignore the `wiretype` and the other arguments here it should be feasible (and easier to grasp) to call `Decode` on `val.Interface` earlier (from the appropriate callee) instead. Sorry, I still need some time to investigate how (and why this works) and if there is a problem

**Comment by ineiti** [comment]
Tested with cothority - it works.

File: decode.go (hunk 1 above)
Comment (Liamsi) [inline decode.go]: OK, I see this was merged now in #29

**Pull request closed without merging**